crates/templates/src/app_info.rs

```diff
@@ -67,15 +72,61 @@ impl AppInfo {
         })
     }
 
+    fn from_existent_template(manifest_path: &Path) -> anyhow::Result<Self> {
+        // This has to be cruder, because (with the v2 style of component) a template
+        // is no longer valid TOML, so `from_existent_file` fails at manifest
+        // version inference.
+        let manifest_tpl_str = std::fs::read_to_string(manifest_path)?;
+
+        // TODO: a better way
+        let is_v1_tpl = manifest_tpl_str.contains("spin_manifest_version = \"1\"");
```

Comment:
Not unless we think someone is going to use Spin 0.9 templates with Spin 2 (or unless there are third party templates out there that haven't been updated since Spin 0.9).  And it defaults to forgiving, so if there _are_ very old templates out there, they will be treated as "allow the add" rather than anything blocking.

On the other hand, yes, I would like this check to be more aligned with the actual TOML parser!